sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -83,14 +84,29 @@ def get_inference_runner(self) -> InferenceRunner:
 
 
 class RunInference(beam.PTransform):
-  """An extensible transform for running inferences."""
-  def __init__(self, model_loader: ModelLoader, clock=None):
+  """An extensible transform for running inferences.
+  Args:
+      model_loader: An implementation of InferenceRunner.
+      clock: A clock implementing get_current_time_in_microseconds.
+      close_to_resource: A string representing the resource location hints.
+  """
+  def __init__(self,
+               model_loader: ModelLoader,
+               clock:_Clock=None,
+               close_to_resource:str=None):
     self._model_loader = model_loader
     self._clock = clock
+    self._close_to_resource = close_to_resource
```

Comment:
That actually makes a lot of sense and I think it will work.

In the TFX-BSL implementation, the data does come from the input paramter which the loader has access to.
https://github.com/tensorflow/tfx-bsl/blob/182918aaefd5287e7669d87bfc818155470315aa/tfx_bsl/beam/run_inference.py#L118